backend/filemanagement-service/src/main/java/hu/me/iit/malus/thesis/filemanagement/controller/FileManagementController.java

```diff
@@ -2,53 +2,42 @@
 
 
 import hu.me.iit.malus.thesis.filemanagement.controller.converters.Converter;
-import hu.me.iit.malus.thesis.filemanagement.controller.dto.File;
-import hu.me.iit.malus.thesis.filemanagement.controller.dto.Service;
+import hu.me.iit.malus.thesis.filemanagement.controller.dto.FileDescriptorDto;
 import hu.me.iit.malus.thesis.filemanagement.model.FileDescription;
+import hu.me.iit.malus.thesis.filemanagement.model.Service;
 import hu.me.iit.malus.thesis.filemanagement.service.FileManagementService;
 import hu.me.iit.malus.thesis.filemanagement.service.exceptions.FileNotFoundException;
 import hu.me.iit.malus.thesis.filemanagement.service.exceptions.UnsupportedOperationException;
+import lombok.RequiredArgsConstructor;
 import org.glassfish.jersey.media.multipart.FormDataParam;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.FileSystemResource;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import javax.servlet.http.Part;
 import javax.ws.rs.FormParam;
 import java.io.IOException;
 import java.security.Principal;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * Controller endpoint of this service, which handles file transfers
+ *
  * @author Ilku Krisztián
+ * @author Attila Szőke
  */
 @RestController
 @RequestMapping("/api/filemanagement")
+@RequiredArgsConstructor
 public class FileManagementController {
 
-    private FileManagementService fileManagementService;
-
-    @Autowired
-    public FileManagementController(FileManagementService fileManagementService) {
-        this.fileManagementService = fileManagementService;
-    }
-
-    @PostMapping("/upload")
-    public ResponseEntity<File> uploadFile(@FormDataParam("file") Part file, @FormParam("service") Service service, @FormParam("tagId") Long tagId, Principal principal) throws IOException {
-        FileDescription fd = fileManagementService.uploadFile(file, service, principal.getName(), tagId);
-        if (fd == null) return ResponseEntity
-                .status(204)
-                .body(null);
-
-        return ResponseEntity
-                .status(200)
-                .body(Converter.FileDescriptionToFile(fd));
-    }
+    private final FileManagementService fileManagementService;
 
     @PostMapping("/uploadFiles")
-    public ResponseEntity<Set<File>> uploadFileMultipleFiles(@FormDataParam("file") ArrayList<Part> file, @FormParam("service") Service service, @FormParam("tagId") Long tagId, Principal principal) throws IOException {
-        Set<File> result = new HashSet<>();
+    public ResponseEntity<Set<FileDescriptorDto>> uploadFiles(@FormDataParam("file") ArrayList<Part> file, @FormParam("service") Service service, @FormParam("tagId") Long tagId, Principal principal) throws IOException {
+        Set<FileDescriptorDto> result = new HashSet<>();
```

Comment:
validacio??? rossz formid, tagid, akkor hol vagodik el? itt kellene

Reply:
Nem volt eszembe mert nem volt az issue része, de tényleg akár már meg is csinálhatom ha egyszer úgyis belenyúlok. Javítva.